RecoLocalTracker/SiPixelRecHits/plugins/SiPixelRecHitFromSOA.cc

```diff
@@ -0,0 +1,196 @@
+#include <cuda_runtime.h>
+
+#include "CUDADataFormats/Common/interface/Product.h"
+#include "CUDADataFormats/Common/interface/HostProduct.h"
+#include "CUDADataFormats/TrackingRecHit/interface/TrackingRecHit2DCUDA.h"
+#include "DataFormats/Common/interface/DetSetVectorNew.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/SiPixelCluster/interface/SiPixelCluster.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHitCollection.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "HeterogeneousCore/CUDACore/interface/ScopedContext.h"
+#include "RecoLocalTracker/SiPixelRecHits/interface/pixelCPEforGPU.h"
+
+class SiPixelRecHitFromSOA : public edm::stream::EDProducer<edm::ExternalWork> {
+public:
+  explicit SiPixelRecHitFromSOA(const edm::ParameterSet& iConfig);
+  ~SiPixelRecHitFromSOA() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  using HMSstorage = HostProduct<unsigned int[]>;
+
+private:
+  void acquire(edm::Event const& iEvent,
+               edm::EventSetup const& iSetup,
+               edm::WaitingTaskWithArenaHolder waitingTaskHolder) override;
+  void produce(edm::Event& iEvent, edm::EventSetup const& iSetup) override;
+
+  const edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> geomToken_;
+
+  edm::EDGetTokenT<cms::cuda::Product<TrackingRecHit2DCUDA>> tokenHit_;  // CUDA hits
+  edm::EDGetTokenT<SiPixelClusterCollectionNew> clusterToken_;           // Legacy Clusters
+
+  uint32_t m_nHits;
+  cms::cuda::host::unique_ptr<uint16_t[]> m_store16;
+  cms::cuda::host::unique_ptr<float[]> m_store32;
+  cms::cuda::host::unique_ptr<uint32_t[]> m_hitsModuleStart;
+};
+
+SiPixelRecHitFromSOA::SiPixelRecHitFromSOA(const edm::ParameterSet& iConfig)
+    : geomToken_(esConsumes()),
+      tokenHit_(
+          consumes<cms::cuda::Product<TrackingRecHit2DCUDA>>(iConfig.getParameter<edm::InputTag>("pixelRecHitSrc"))),
+      clusterToken_(consumes<SiPixelClusterCollectionNew>(iConfig.getParameter<edm::InputTag>("src"))) {
+  produces<SiPixelRecHitCollectionNew>();
```

Comment:
given https://github.com/cms-sw/cmssw/blob/1e4db77b3aa34787e841604d7d89c460ae112fc7/DataFormats/TrackerRecHit2D/interface/SiPixelRecHitCollection.h#L7-L9

does it make sense to use `SiPixelRecHitCollectionNew` ?
Why not simply `SiPixelRecHitCollection` ?

Reply:
sounds like a good time to get rid of the `New`

Reply:
a name migration 12 years in the making https://github.com/cms-sw/cmssw/commit/2be7a937d37494ba98d9810409e6389a4fb47113

Reply:
changed `SiPixelRecHitCollectionNew` to `SiPixelRecHitCollection` in https://github.com/cms-patatrack/cmssw/pull/593/commits/6a4e3e5985bcb32b27010346995e3631c50cd40e

Reply:
History, of course is present in many other places